core/state/trie_prefetcher.go

```diff
@@ -190,7 +225,11 @@ func (p *triePrefetcher) trie(owner common.Hash, root common.Hash) Trie {
 // used marks a batch of state items used to allow creating statistics as to
 // how useful or wasteful the prefetcher is.
 func (p *triePrefetcher) used(owner common.Hash, root common.Hash, used [][]byte) {
-	if fetcher := p.fetchers[p.trieID(owner, root)]; fetcher != nil {
+	p.fetchersLock.Lock()
```

Comment:
The purpose of fetchersLock is to protect the fetchers map from concurrent read and write operations. In this case, fetchers is only being read and not written, so perhaps an RLock is sufficient.
As for the change `fetcher.used = used`, it should not be within the scope of the lock's protection.

Reply:
It sounds good! Applied.